dockerfiles/che/launcher.sh

```diff
@@ -0,0 +1,380 @@
+}
+
+get_che_hostname() {
+  INSTALL_TYPE=$(get_docker_install_type)
+  CHE_IP=$(get_docker_host_ip)
+
+  if [ "${INSTALL_TYPE}" = "boot2docker" ]; then
+    echo "${CHE_IP}"
+  elif [ "${INSTALL_TYPE}" = "moby" ] && [ "${CHE_IP}" = "10.0.75.2" ]; then
+    echo "${CHE_IP}"
+  else
+    echo "localhost"
+  fi
+}
+
+check_docker() {
+  if [ ! -S /var/run/docker.sock ]; then
+    error_exit "Docker socket (/var/run/docker.sock) hasn't be mounted \
+inside the container. Verify the syntax of the \"docker run\" command."
+  fi
+
+  if ! docker ps > /dev/null 2>&1; then
+    output=$(docker ps)
+    error_exit "Error when running \"docker ps\": ${output}"
+  fi
+}
+
+che_container_exist() {
+  if [ "$(docker ps -aq  -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_running() {
+  if [ "$(docker ps -qa -f "status=running" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_stopped() {
+  if [ "$(docker ps -qa -f "status=exited" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+wait_until_container_is_running() {
+  CONTAINER_START_TIMEOUT=${1}
+
+  ELAPSED=0
+  until che_container_is_running || [ ${ELAPSED} -eq "${CONTAINER_START_TIMEOUT}" ]; do
+    sleep 1
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+server_is_booted() {
+  if ! curl -v http://"${CHE_HOST_IP}":"${CHE_PORT}"/api > /dev/null 2>&1 ; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+wait_until_server_is_booted () {
+  SERVER_BOOT_TIMEOUT=${1}
+
+  ELAPSED=0
+  until server_is_booted || [ ${ELAPSED} -eq "${SERVER_BOOT_TIMEOUT}" ]; do
+    sleep 1
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+parse_command_line () {
+  if [ $# -eq 0 ]; then
+    usage
+    exit
+  fi
+
+  for command_line_option in "$@"; do
+    case ${command_line_option} in
+      start|stop|restart|update|info)
+        CHE_SERVER_ACTION=${command_line_option}
+      ;;
+      -h|--help)
+        usage
+        exit
+      ;;
+      *)
+        # unknown option
+        error_exit "You passed an unknown command line option."
+      ;;
+    esac
+  done
+}
+
+start_che_server() {
+  if che_container_exist; then
+    error_exit "A container named \"${CHE_SERVER_CONTAINER_NAME}\" already exists. Please remove it manually (docker rm -f ${CHE_SERVER_CONTAINER_NAME}) and try again."
+  fi
+
+  CURRENT_IMAGE=$(docker images -q "${CHE_SERVER_IMAGE_NAME}":"${CHE_VERSION}")
+
+  if [ "${CURRENT_IMAGE}" != "" ]; then
+    info "ECLIPSE CHE: ALREADY HAVE IMAGE ${CHE_SERVER_IMAGE_NAME}:${CHE_VERSION}"
+  else
+    update_che_server
+  fi
+
+  info "ECLIPSE CHE: CONTAINER STARTING"
+  docker run -d --name "${CHE_SERVER_CONTAINER_NAME}" \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v "${CHE_DATA_FOLDER}"/lib:/home/user/che/lib-copy \
+    -v "${CHE_DATA_FOLDER}"/workspaces:/home/user/che/workspaces \
```

Comment:
Looks like syntax for running che has been changed recently. This mount is not needed. And 2 additional env variable are needed. Please check master.